### ISSUE TITLE
Unable to reinitialize agent once shut downed

### ISSUE THREAD

**Issue by GHkrishna** [openwallet-foundation/credo-ts#2135]
## What:
Reinitializing agent with `await agent.initialize()` after `await agent.shutdown()` gives an indyVdr error due to this implementation. Should we just return instead of throwing error:
```typescript
public connect() {
    if (this._pool) {
      throw new IndyVdrError('Cannot connect to pool, already connected.')
    }

    this._pool = new PoolCreate({
      parameters: {
        transactions: this.config.genesisTransactions,
      },
    })
  }
```

This becomes,
```typescript
public connect() {
    if (this._pool) {
      return
    }

    this._pool = new PoolCreate({
      parameters: {
        transactions: this.config.genesisTransactions,
      },
    })
  }
```
## To reproduce
1. await agent.shutdown()
2. await agent.initialize()

## Version 
"@credo-ts/indy-vdr": "0.5.13"

**Comment by TimoGlastra** [comment]
I think the better approach would be to shutdown the pool at agent shutdown and remove the value initialization

**Comment by GHkrishna** [comment]
The issue for shutting down of pool still exists in IndyVDR. For now I think we can remove the initialized value and set it to null. And once it is resolved in VDR we can retro fit it. What do you think?

**Comment by TimoGlastra** [comment]
Yes i think that makes sense. But since we can't close it we should reuse the existing pool on initialize again

**Comment by GHkrishna** [comment]
Closing as completed by https://github.com/openwallet-foundation/credo-ts/pull/2137
